// hamburger-react/dist-cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Cross", {
  enumerable: true,
  get: function () {
    return _Cross.Cross;
  }
});
Object.defineProperty(exports, "Divide", {
  enumerable: true,
  get: function () {
    return _Divide.Divide;
  }
});
Object.defineProperty(exports, "Fade", {
  enumerable: true,
  get: function () {
    return _Fade.Fade;
  }
});
Object.defineProperty(exports, "Pivot", {
  enumerable: true,
  get: function () {
    return _Pivot.Pivot;
  }
});
Object.defineProperty(exports, "Rotate", {
  enumerable: true,
  get: function () {
    return _Rotate.Rotate;
  }
});
Object.defineProperty(exports, "Slant", {
  enumerable: true,
  get: function () {
    return _Slant.Slant;
  }
});
Object.defineProperty(exports, "Sling", {
  enumerable: true,
  get: function () {
    return _Sling.Sling;
  }
});
Object.defineProperty(exports, "Spin", {
  enumerable: true,
  get: function () {
    return _Spin.Spin;
  }
});
Object.defineProperty(exports, "Spiral", {
  enumerable: true,
  get: function () {
    return _Spiral.Spiral;
  }
});
Object.defineProperty(exports, "Squash", {
  enumerable: true,
  get: function () {
    return _Squash.Squash;
  }
});
Object.defineProperty(exports, "Squeeze", {
  enumerable: true,
  get: function () {
    return _Squeeze.Squeeze;
  }
});
Object.defineProperty(exports, "Turn", {
  enumerable: true,
  get: function () {
    return _Turn.Turn;
  }
});
Object.defineProperty(exports, "Twirl", {
  enumerable: true,
  get: function () {
    return _Twirl.Twirl;
  }
});
exports.default = void 0;

var _Tilt = require("./Tilt");

var _Cross = require("./Cross");

var _Divide = require("./Divide");

var _Fade = require("./Fade");

var _Pivot = require("./Pivot");

var _Rotate = require("./Rotate");

var _Slant = require("./Slant");

var _Sling = require("./Sling");

var _Spin = require("./Spin");

var _Spiral = require("./Spiral");

var _Squash = require("./Squash");

var _Squeeze = require("./Squeeze");

var _Turn = require("./Turn");

var _Twirl = require("./Twirl");

var _default = _Tilt.Tilt;
exports.default = _default;
